dynamic-property-api/src/main/java/ru/fix/dynamic/property/source/SourcedProperty.java

```diff
@@ -0,0 +1,82 @@
+package ru.fix.dynamic.property.source;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.fix.dynamic.property.api.DynamicProperty;
+import ru.fix.dynamic.property.api.DynamicPropertyListener;
+import ru.fix.dynamic.property.api.DynamicPropertySource;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Contain property initial value. Automatically register property change listener.
+ */
+public class SourcedProperty<T> implements DynamicProperty<T> {
+
+    private static final Logger log = LoggerFactory.getLogger(SourcedProperty.class);
+
+    private DynamicPropertySource propertySource;
+    private Class<T> type;
+    private String name;
+    private T defaultValue;
+    private volatile T currentValue;
+
+    private List<DynamicPropertyListener<T>> listeners = new CopyOnWriteArrayList<>();
+
+    public SourcedProperty(DynamicPropertySource propertySource, String name, Class<T> type) {
+        this(propertySource, name, type, null);
+    }
+
+    public SourcedProperty(DynamicPropertySource propertySource, String name, Class<T> type, T defaultValue) {
+        this.propertySource = propertySource;
+        this.name = name;
+        this.type = type;
+        this.defaultValue = defaultValue;
+
+        init();
+    }
+
+    private void init() {
+        propertySource.addPropertyChangeListener(
```

Comment:
This approach with subscription and getting current value in different steps are error prone. 
Suggestion: let's change PropertySource method 
`<T> void addPropertyChangeListener(String propertyName, Class<T> type, DynamicPropertyListener<T> listener);` to
`<T> T addPropertyChangeListener(String propertyName, Class<T> type, DynamicPropertyListener<T> listener);`
and implement this method in a way that subscription and getting current value for user point of view looks like atomic. And there will be no missing update here.

Reply:
decided not to do it